x/evm/keeper/grpc_query.go

```diff
@@ -467,6 +494,39 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
 	}, nil
 }
 
+// GetTxTraceResultForBlock returns TxTraceResult and
+// statedb with cached address list when need patch and
+func (k Keeper) GetTxTraceResultForBlock(
+	ctx sdk.Context,
+	tx *types.MsgEthereumTx,
+	signer ethtypes.Signer,
+	cfg *statedb.EVMConfig,
+	txConfig statedb.TxConfig,
+	traceConfig *types.TraceConfig,
+	patch bool,
+	lastDB *statedb.StateDB,
+) (*statedb.StateDB, *types.TxTraceResult) {
+	result := new(types.TxTraceResult)
+	ethTx := tx.AsTransaction()
+	txConfig.TxHash = ethTx.Hash()
+	var stateDB *statedb.StateDB
+	if patch {
+		stateDB = statedb.New(ctx, &k, txConfig)
+		if lastDB != nil {
+			stateDB.SetAddressToAccessList(lastDB.GetAddressToAccessList())
+		}
+		lastDB = stateDB
+	}
+	traceResult, logIndex, err := k.traceTx(ctx, cfg, txConfig, stateDB, signer, ethTx, traceConfig, true, nil)
+	if err != nil {
+		result.Error = err.Error()
+	} else {
+		txConfig.LogIndex = logIndex
```

Comment:
Why assign `txConfig.LogIndex` here when it doesn't get returned or used in the rest of the function?